lib/coinbase/address.rb

```diff
@@ -194,5 +159,56 @@ def addresses_api
     def transfers_api
       @transfers_api ||= Coinbase::Client::TransfersApi.new(Coinbase.configuration.api_client)
     end
+
+    def destination_address_and_network(destination)
+      return [destination.default_address.id, destination.network_id] if destination.is_a?(Wallet)
+      return [destination.id, destination.network_id] if destination.is_a?(Address)
+
+      [destination, network_id]
+    end
+
+    def validate_can_transfer!(amount, asset_id, destination_network_id)
+      raise 'Cannot transfer from address without private key loaded' if @key.nil?
+
+      raise ArgumentError, "Unsupported asset: #{asset_id}" unless Coinbase::Asset.supported?(asset_id)
+
+      raise ArgumentError, 'Transfer must be on the same Network' unless destination_network_id == network_id
+
+      current_balance = balance(asset_id)
+
+      return unless current_balance < amount
+
+      raise ArgumentError, "Insufficient funds: #{amount} requested, but only #{current_balance} available"
+    end
+
+    def create_transfer(amount, asset_id, destination)
+      create_transfer_request = {
+        amount: Coinbase::Asset.to_atomic_amount(amount, asset_id).to_i.to_s,
+        network_id: network_id,
+        asset_id: Coinbase::Asset.primary_denomination(asset_id).to_s,
+        destination: destination
+      }
+
+      transfer_model = Coinbase.call_api do
+        transfers_api.create_transfer(wallet_id, id, create_transfer_request)
+      end
+
+      Coinbase::Transfer.new(transfer_model)
+    end
+
+    def sign_transfer(transfer)
```

Comment:
could you add docs for these new methods and document what they return?

Reply:
These are private methods, we can add docs but we would explicitly want them not to be included